File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // set<int> set;
        // for(int i = 0; i < nums.size(); i++){
        //     set.insert(nums[i]);
        // }
        // int index = 0; 
        // for(auto it : set){
        //     nums[index] = it;
        //     index++;
        // }
        // return set.size();

        int i = 0;
        for(int j = 1; j < nums.size(); j++){
            if(nums[j] != nums[i]){
                nums[i + 1] = nums[j];
                i++;
            }
        }
        return i + 1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int i = 0; i < nums.length; i++){
            set.add(nums[i]);
        }
        List<Integer> list = new ArrayList<>(set);
        Collections.sort(list);
        int k = set.size(), j = 0;
        for(int x : list){
            nums[j++] = x;
        }
        return k;
    }
}